examples/llm/benchmarks/plot_pareto.py

```diff
@@ -0,0 +1,216 @@
+            ],
+            "is_pareto_efficient": False,
+        }
+        for result in results
+    ]
+    df = pd.DataFrame(data_points)
+
+    def pareto_efficient(ids, points):
+        """
+        Mark Pareto-efficient points.
+        A point p is dominated if there's another q
+        such that q is >= p in all dimensions.
+        """
+        points = np.array(points)
+        pareto_points = []
+        for i, (point_id, point) in enumerate(zip(ids, points)):
+            dominated = False
+            for j, other_point in enumerate(points):
+                if i != j and all(other_point >= point):
+                    dominated = True
+                    break
+            if not dominated:
+                pareto_points.append(point)
+                df.at[point_id, "is_pareto_efficient"] = True
+        return np.array(pareto_points)
+
+    sns.set(style="whitegrid")
+    fig, ax = plt.subplots(figsize=(14, 6), constrained_layout=True)
+
+    labels = df["label"].unique()
+
+    for label in labels:
+        group = df[df["label"] == label]
+        # Scatter all points
+        ax.scatter(
+            group[f"output_token_throughput_per_request_{stat_value}"],
+            group["output_token_throughput_per_gpu_avg"],
+            label=f"Label {label}",
+        )
+
+        # Identify and mark Pareto frontier
+        pareto_points = pareto_efficient(
+            group.index,
+            group[
+                [
+                    f"output_token_throughput_per_request_{stat_value}",
+                    "output_token_throughput_per_gpu_avg",
+                ]
+            ].values,
+        )
+        # Sort by x-value for a clean line
+        pareto_points = pareto_points[np.argsort(pareto_points[:, 0])]
+        ax.plot(
+            pareto_points[:, 0],
+            pareto_points[:, 1],
+            linestyle="--",
+            label=f"Pareto Frontier {label}",
+        )
+
+    # Save CSV
+    if stat_value == "avg":
+        df_file_name = "results.csv"
+    else:
+        df_file_name = f"results_{stat_value}.csv"
+    df.to_csv(df_file_name)
+
+    # Axis labels and tick intervals
+    ax.set_xlabel(f"tokens/s/user {stat_value}")
+    ax.set_ylabel("tokens/s/gpu avg")
+    ax.set_title(f"Pareto - {title}")
+    ax.legend(bbox_to_anchor=(1.02, 1), loc="upper left")
+
+    ax.grid(True)
+    x_interval = 5
+    y_interval = 5
+    ax.xaxis.set_major_locator(MultipleLocator(x_interval))
+    ax.yaxis.set_major_locator(MultipleLocator(y_interval))
+
+    if stat_value == "avg":
+        file_name = "pareto_plot.png"
+    else:
+        file_name = f"pareto_plot_{stat_value}.png"
+    plt.savefig(file_name, dpi=300)
+    plt.close()
+
+
+if __name__ == "__main__":
+    genai_perf_profile_export_json_paths = get_genai_perf_profile_export_json_paths(
+        [
+            "artifacts_vllm_serve_tp4dp2",
+            "artifacts_disagg_prefill_tp1dp4_decode_tp4dp1",
```

Comment:
Will this work out of box? I think the instructions just say to "run in same directory as `artifacts/` directory". Should there be an extra step in docs to do some `mkdir`s and `mv artifacts ...` to follow this structure and auto-update the labels on the plot/legend?

Reply:
Nvm - there are some worded steps about this.